pandas/tests/indexes/test_category.py

```diff
@@ -507,6 +507,22 @@ def test_identical(self):
         self.assertTrue(ci1.identical(ci1.copy()))
         self.assertFalse(ci1.identical(ci2))
 
+    def test_ensure_copied_data(self):
+        # Check the "copy" argument of each Index.__new__ is honoured
+        # GH12309
+        # Must be tested separately from other indexes because
+        # self.value is not an ndarray
+        for index in self.indices.values():
+            result = CategoricalIndex(index.values, copy=True)
+            tm.assert_index_equal(index, result)
+            if index.values.base is not None:
+                self.assertIsNot(index.values.base, self.result.base)
+
+            result = CategoricalIndex(index.values, copy=False)
+            bases = [i.values if i.values.base is None else i.values.base
```

Comment:
No: I previously did but [now that function ensures the arguments are really ndarrays](https://github.com/pydata/pandas/commit/c2ea8fb20611370c7db7e11506727b0b67a76662#diff-9d7323cc6b0cfa3f4482bc8d4268966dR986) (understandably). This is why this index now requires a special test. 

(Except if you prefer to not use ``tm.assert_numpy_array_equal`` at all, even in the generic test, and merge this in it. For the records: the special test for ``MultiIndex`` - my next PR - will still use ``tm.assert_numpy_array_equal(......check_same=?)``)

Done the rest.